pkg/shares/non_interactive_defaults.go

```diff
@@ -83,12 +86,54 @@ func roundUpBy(cursor, v int) int {
 	}
 }
 
-// MinSquareSize returns the minimum square size that can contain shareCount
+// BlobMinSquareSize returns the minimum square size that can contain shareCount
 // number of shares.
-func MinSquareSize(shareCount int) int {
+func BlobMinSquareSize(shareCount int) int {
 	return RoundUpPowerOfTwo(int(math.Ceil(math.Sqrt(float64(shareCount)))))
 }
 
+// SubTreeWidth determines the number of leaves per subtree in the share
+// commitment over a given blob. The input should be the total number of shares
+// used by that blob. The reasoning behind this algorithm is discussed in depth
+// in ADR013
+// (celestia-app/docs/architecture/adr-013-non-interative-default-rules-for-zero-padding).
+func SubTreeWidth(shareCount int) int {
+	// per ADR013, we use a predetermined threshold to determine width of sub
+	// trees used to create share commitments
+	s := (shareCount / appconsts.SubtreeRootThreshold)
+
+	// round up if the width is not an exact multiple of the threshold
+	if shareCount%appconsts.SubtreeRootThreshold != 0 {
+		s++
+	}
+
+	// use a power of two equal to or larger than the multiple of the subtree
+	// root threshold
+	s = RoundUpPowerOfTwo(s)
+
+	// use the minimum of the subtree width and the min square size, this
+	// gurarantees that a valid value is returned
+	s = min(s, BlobMinSquareSize(shareCount))
+
+	// using a value below the min square size is wasteful, see ADR013 for more
+	// details
+	return max(s, appconsts.DefaultMinSquareSize)
+}
```

Comment:
note the update to the algo per the call this morning:

the blob min square size was re-added, then we're using the min of that value.